plugins/modules/domain.py

```diff
@@ -64,25 +64,34 @@
 )
 
 
-def validate_record(existing_records, state, client,
+def validate_record(existing_records, state, client, module,
```

Comment:
I don't get why you need `module` here, you don't seem to be using it in the function are you ?